visualization/app/codeCharta/ui/attributeSideBar/attributeSideBar.component.spec.ts

```diff
@@ -1,292 +1,35 @@
-		})
-
-		it("should call function to update secondaryMetricKeys", () => {
-			attributeSideBarController.onEdgeMetricChanged(edgeMetric)
-
-			expect(attributeSideBarController["updateSortedMetricKeysWithoutPrimaryMetrics"]).toHaveBeenCalled()
-		})
-	})
-
-	describe("onClickCloseSideBar", () => {
-		it("should call set isAttributeSideBarVisible to false", () => {
-			storeService.dispatch(openAttributeSideBar())
-
-			attributeSideBarController.onClickCloseSideBar()
-
-			expect(storeService.getState().appSettings.isAttributeSideBarVisible).toBeFalsy()
+describe("AttributeSideBarComponent", () => {
+	beforeEach(() => {
+		TestBed.configureTestingModule({
+			imports: [AttributeSideBarModule]
 		})
 	})
 
-	describe("onClickNodeName", () => {
-		it("should open file if node is a leaf", () => {
-			attributeSideBarController["_viewModel"].node = klona(TEST_NODE_LEAF)
-
-			LazyLoader.openFile = jest.fn()
-			attributeSideBarController.onClickNodeName()
+	it("should not display side bar if there is no node selected", async () => {
+		const { container } = await render(AttributeSideBarComponent, { excludeComponentDeclaration: true })
 
-			expect(LazyLoader.openFile).toHaveBeenCalled()
-		})
+		expect(container.querySelector(".expanded")).toBe(null)
 	})
 
-	describe("updateSortedMetricKeysWithoutPrimaryMetrics", () => {
-		beforeEach(() => {
-			attributeSideBarController["_viewModel"].node = klona(TEST_NODE_LEAF)
-			attributeSideBarController["_viewModel"].node.attributes = {
-				a: 1,
-				b: 2,
-				c: 3,
-				d: 4,
-				e: 5
-			}
-		})
-
-		it("should filter node.attributes by selected metricKeys 1", () => {
-			attributeSideBarController["_viewModel"].primaryMetricKeys = {
-				node: {
-					area: "a"
-				}
-			} as PrimaryMetrics
-
-			attributeSideBarController["updateSortedMetricKeysWithoutPrimaryMetrics"](false)
-
-			const expectedSecondaryMetrics = [
-				{ name: "b", type: undefined },
-				{ name: "c", type: undefined },
-				{ name: "d", type: undefined },
-				{ name: "e", type: undefined }
-			]
-
-			expect(attributeSideBarController["_viewModel"].secondaryMetrics).toEqual(expectedSecondaryMetrics)
-		})
-
-		it("should filter node.attributes by selected metricKeys 2", () => {
-			attributeSideBarController["_viewModel"].primaryMetricKeys = {
-				node: {
-					area: "a",
-					color: "c",
-					height: "e"
-				}
-			} as PrimaryMetrics
-
-			attributeSideBarController["updateSortedMetricKeysWithoutPrimaryMetrics"](false)
-
-			const expectedSecondaryMetrics = [
-				{ name: "b", type: undefined },
-				{ name: "d", type: undefined }
-			]
-
-			expect(attributeSideBarController["_viewModel"].secondaryMetrics).toEqual(expectedSecondaryMetrics)
-		})
-	})
+	// test display filename
+	// test primary and secondary metrics (sorted), with attribute type change
+	// test close side bar
+	// test click on leaf calls LazyLoader.openFile = jest.fn()
+	// TEST_NODE_LEAF
```

Comment:
I guess this is something that is still worked upon?

Reply:
done